programs/drift/src/instructions/keeper.rs

```diff
@@ -498,43 +510,44 @@ pub fn place_swift_taker_order<'c: 'info, 'info>(
         panic!("Swift orders are disabled on mainnet-beta");
     }
 
-    // Authenticate the signature
+    // Authenticate the swift param message
     let ix_idx = load_current_index_checked(ix_sysvar)?;
     validate!(
-        ix_idx > 0,
+        ix_idx > 1,
         ErrorCode::InvalidVerificationIxIndex,
-        "instruction index must be greater than 0"
+        "instruction index must be greater than 1 for two sig verifies"
+    )?;
+    let ix: Instruction = load_instruction_at_checked(ix_idx as usize - 2, ix_sysvar)?;
+    verify_ed25519_ix(
+        &ix,
+        &swift_server::id().to_bytes(),
+        &swift_message.clone().try_to_vec()?,
+        &sig,
     )?;
+
     let ix: Instruction = load_instruction_at_checked(ix_idx as usize - 1, ix_sysvar)?;
     verify_ed25519_ix(
         &ix,
         &taker.authority.to_bytes(),
         &taker_order_params_message.clone().try_to_vec()?,
-        &sig,
+        &swift_message.swift_order_signature.clone(),
```

Comment:
do you need to clone if it is a reference?

Reply:
removed unnecessary clone